frontend/src/pages/FundsPullPreApprovals.tsx

```diff
@@ -0,0 +1,103 @@
+// Copyright (c) The Diem Core Contributors
+// SPDX-License-Identifier: Apache-2.0
+
+import React, { useEffect, useState } from "react";
+import { Approval } from "../interfaces/approval";
+import BackendClient from "../services/backendClient";
+import TestnetWarning from "../components/TestnetWarning";
+import Breadcrumbs from "../components/Breadcrumbs";
+import { Container } from "reactstrap";
+import FundsPullPreApprovalsList from "../components/FundsPullPreApproval/FundsPullPreApprovalsList";
+
+const REFRESH_APPROVALS_INTERVAL = 3000;
+
+function FundsPullPreApprovals() {
+  const [newApprovals, setNewApprovals] = useState<Approval[]>([]);
+  const [activeApprovals, setActiveApprovals] = useState<Approval[]>([]);
+  const [historyApprovals, setHistoryApprovals] = useState<Approval[]>([]);
+
+  let refreshApprovals = true;
+
+  useEffect(() => {
+    const fetchApprovals = async () => {
+      try {
+        if (refreshApprovals) {
+          let innerNewApprovals: Approval[] = [];
+          let innerActiveApprovals: Approval[] = [];
+          let innerHistoryApprovals: Approval[] = [];
+          const approvals = await new BackendClient().getAllFundsPullPreApprovals();
+          for (const approval of approvals) {
+            if (approval.status === "pending") {
+              innerNewApprovals.push(approval);
+            } else if (approval.status === "valid") {
+              const now = new Date().toISOString();
+              if (Date.parse(approval!.scope.expiration_timestamp) < Date.parse(now)) {
+                innerHistoryApprovals.push(approval);
+              } else {
+                innerActiveApprovals.push(approval);
+              }
+            } else {
+              innerHistoryApprovals.push(approval);
+            }
+          }
+          setNewApprovals(innerNewApprovals);
+          setActiveApprovals(innerActiveApprovals);
+          setHistoryApprovals(innerHistoryApprovals);
+        }
+        setTimeout(fetchApprovals, REFRESH_APPROVALS_INTERVAL);
+      } catch (e) {
+        console.error(e);
+      }
+    };
+
+    // noinspection JSIgnoredPromiseFromCall
+    fetchApprovals();
+
+    return () => {
+      refreshApprovals = false;
+    };
+  }, []);
```

Comment:
you should specify as a dependency all the external variables used within the effect. For example `setNewApprovals` `setActiveApprovals`

Reply:
@vladkucherov - only the setters or the variables as well?